internal/common/utilities_ymlescape.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+)
+
+const (
+	bsCode                           byte = byte('\\')
+	metaCharactersNeedsEscapePattern      = `.*[.+*?()|[\]{}^$].*`
+)
+
+var metaRegex = regexp.MustCompile(metaCharactersNeedsEscapePattern)
+
+type YmlEscapeHandlers struct{}
+
+// Escape function is required, as yaml library no longer maintained
+// yaml unmaintained library issue https://github.com/go-yaml/yaml/pull/862
```

Comment:
Please check if the sig yaml library (sigs.k8s.io/yaml, which contains a copy of the go-yaml) also still has this issue.

Currently looking for a way to fully migrate to the sig yaml library.

Reply:
sigs.k8s.io is just a wrapper unfortunately. 

![Screenshot 2025-01-06 at 21 43 12](https://github.com/user-attachments/assets/4a322e85-08af-4b4a-99dd-ea0396f3730d)

I was thinking about forking `gopkg.in/yaml.v3` as an option. Not too much work

Reply:
This yaml.Node is another problem )). There are few dependencies, they are super old, and they rely on yaml.v3 structs.... So it's like ripping off bones.

With this pull request, main target was to cover functionality with tests. So there is quite a decent coverage. Should simplify moving to another library when ready. I have no solution at the mean time unfortunately.

Reply:
> Please check if the sig yaml library (sigs.k8s.io/yaml, which contains a copy of the go-yaml) also still has this issue.

I was not able to make it work, due to other dependencies explicitly require `gopkg.in/yaml.v3.Node` for example.

What do you think? Shell we merge this one in, and you will try to swap libraries, this will require some rewrite to how helm-unittest does yaml node traversing? With this pull request we going to have a test coverage that new library need to satisfy.

I could fork `"gopkg.in/yaml.v3` and fix it there, as alternative solution.

Reply:
This is good for the goal we want to achieve.

About forking or using another yaml library, I'm Currently looking closely to which library helm is using. This to prevent a different outcome.

Greetings,
@quintush